java/connector-node/risingwave-sink-deltalake/src/main/java/com/risingwave/connector/DeltaLakeSink.java

```diff
@@ -88,13 +87,11 @@ public void write(Iterator<SinkRow> rows) {
                     GenericRecord record = new GenericData.Record(this.sinkSchema);
                     for (int i = 0; i < this.sinkSchema.getFields().size(); i++) {
                         Object values;
-                        if (row.get(i) instanceof Timestamp) {
-                            values = ((Timestamp) row.get(i)).getTime();
-                        } else if (row.get(i) instanceof java.sql.Date) {
+                        if (row.get(i) instanceof LocalDateTime) {
                             values =
-                                    ChronoUnit.DAYS.between(
-                                            LocalDate.ofEpochDay(0),
-                                            ((java.sql.Date) row.get(i)).toLocalDate());
+                                    ((LocalDateTime) row.get(i))
+                                            .toInstant(ZoneOffset.UTC)
+                                            .toEpochMilli();
```

Comment:
cc @xxhZs regarding #13420 #13677 I am not sure whether we need to test for all 4 `LocalDate`/`LocalTime`/`LocalDateTime`/`OffsetDateTime` out of `Object` ...

https://github.com/delta-io/delta/blob/master/PROTOCOL.md#timestamp-without-timezone-timestampntz
Also note that:
* RIsingWave `timestamp` is DeltaLake `TimestampNtz` (`isAdjustedToUTC=false`)
* RisingWave `timestamptz` is DeltaLake `Timestamp` (`isAdjustedToUTC=true`)

Reply:
https://github.com/risingwavelabs/risingwave/pull/14277